scrapy/statscollectors.py

```diff
@@ -28,9 +28,15 @@ def __init__(self, crawler: Crawler):
         self._crawler: Crawler = crawler
 
     def __getattribute__(self, name):
+        cached_name = f"_cached_{name}"
+        try:
+            return super().__getattribute__(cached_name)
```

Comment:
Til `hasattr()` "is implemented by calling getattr(object, name) and seeing whether it raises an AttributeError or not"